src/main/scala/mesosphere/marathon/state/RootGroup.scala

```diff
@@ -57,7 +57,7 @@ class RootGroup(
       app <- group.apps.values
       dependencyId <- app.dependencies
       dependentApp = this.app(dependencyId).map(Set(_))
-      dependentGroup = transitiveGroupsById.get(dependencyId).map(_.transitiveApps)
+      dependentGroup = transitiveGroupsById.get(dependencyId.asAbsolutePath).map(_.transitiveApps)
```

Comment:
Have you thought about changing the dependencies in app to an AbsolutePath as well? Probably shouldn't happen in this PR, but having all Pathes in the state classes absolute wouldn't be bad, right?

Reply:
I'm a little unsure about apps and would like to address that in a separate pull request.